ozark/src/main/java/org/glassfish/ozark/jersey/OzarkFeature.java

```diff
@@ -87,27 +101,47 @@ public void configure(FeatureContext context) {
         if (config.isRegistered(ViewResponseFilter.class)) {
             return;     // already registered!
         }
-        final boolean enableOzark = config.getClasses().stream().anyMatch(this::isController)
-                || config.getInstances().stream().map(o -> o.getClass()).anyMatch(this::isController);
-        if (enableOzark) {
-            context.register(ViewRequestFilter.class);
-            context.register(ViewResponseFilter.class);
-            context.register(ViewableWriter.class);
-            context.register(BindingInterceptorImpl.class);
-            context.register(OzarkModelProcessor.class);
-            context.register(CsrfValidateInterceptor.class);
-            context.register(CsrfProtectFilter.class);
-
-            // Initialize application config object in Mvc class
-            final BeanManager bm = CDI.current().getBeanManager();
-            final MvcContextImpl mvc = (MvcContextImpl) newBean(bm, MvcContextImpl.class);
-            mvc.setConfig(config);
-            mvc.setContextPath(servletContext.getContextPath());
+        Set<Class<?>> controllers = Stream.concat(config.getClasses().stream(), config.getInstances().stream().map(o -> o.getClass()))
+                .filter(this::isController).collect(Collectors.toSet());
+        if (controllers.isEmpty() || hasUnsupportedTypes(controllers)) {
+            return;
         }
+        context.register(ViewRequestFilter.class);
+        context.register(ViewResponseFilter.class);
+        context.register(ViewableWriter.class);
+        context.register(BindingInterceptorImpl.class);
+        context.register(OzarkModelProcessor.class);
+        context.register(CsrfValidateInterceptor.class);
+        context.register(CsrfProtectFilter.class);
+
+        // Initialize application config object in Mvc class
+        final BeanManager bm = CDI.current().getBeanManager();
+        final MvcContextImpl mvc = newBean(bm, MvcContextImpl.class);
+        mvc.setConfig(config);
+        mvc.setContextPath(servletContext.getContextPath());
     }
 
-    private boolean isController(Class<?> c) {
+    boolean isController(Class<?> c) {
         return getAnnotation(c, Controller.class) != null ||
                 Arrays.asList(c.getMethods()).stream().anyMatch(m -> getAnnotation(m, Controller.class) != null);
     }
+
+    /**
+     * <p>MVC classes are required to be CDI-managed beans only.
+     * Managed Beans or EJBs are not allowed (MCV-Spec ch. 2.1.1).</p>
+     */
+    boolean hasUnsupportedTypes(Set<Class<?>> controllers) {
```

Comment:
What about changing the signature to:

```java
boolean hasUnsupportedTypes(Class<?> controller)
```

This way the method would be simpler to test. And you could check if there are problematic controllers and log their names simply by using something like:

```java
controllers.stream()
    .filter( c -> hasUnsupportedTypes(c) )
    .forEach( c -> LOG.server( "...." + c.getSimpleName() ) );
```

Of cause this would create one log entry for each controller, but that's IMO even better compared to the comma separated list of class names.